browser/main/modals/PreferencesModal/BackupTab.js

```diff
@@ -0,0 +1,248 @@
+    this.handleSettingDone = () => {
+      this.setState({BackupAlert: {
+        type: 'success',
+        message: i18n.__('Successfully applied!')
+      }})
+    }
+    this.handleSettingError = (err) => {
+      this.setState({BackupAlert: {
+        type: 'error',
+        message: err.message != null ? err.message : i18n.__('An error occurred!')
+      }})
+    }
+    ipc.addListener('APP_SETTING_DONE', this.handleSettingDone)
+    ipc.addListener('APP_SETTING_ERROR', this.handleSettingError)
+  }
+
+  componentWillUnmount () {
+    ipc.removeListener('APP_SETTING_DONE', this.handleSettingDone)
+    ipc.removeListener('APP_SETTING_ERROR', this.handleSettingError)
+  }
+
+  handleSaveImport (e) {
+    const newConfig = {
+      hotkey: this.state.config.hotkey,
+      ui: this.state.config.ui,
+      editor: this.state.config.editor,
+      preview: this.state.config.preview,
+      blog: this.state.config.blog
+    }
+
+    ConfigManager.set(newConfig)
+
+    store.dispatch({
+      type: 'SET_UI',
+      config: newConfig
+    })
+    this.clearMessage()
+    this.props.haveToSave()
+  }
+
+  handleImportPathBrowseButtonClick (e) {
+    browseFolder(false)
+      .then((targetPath) => {
+        if (targetPath.length > 0) {
+          const { newImport } = this.state
+          newImport.path = targetPath
+          this.setState({
+            newImport
+          })
+        }
+      })
+      .catch((err) => {
+        console.error('BrowseFAILED')
+        console.error(err)
+      })
+  }
+
+  handleExportPathBrowseButtonClick (e) {
+    browseFolder(true)
+      .then((targetPath) => {
+        if (targetPath.length > 0) {
+          const { newExport } = this.state
+          newExport.path = targetPath
+          this.setState({
+            newExport
+          })
+        }
+      })
+      .catch((err) => {
+        console.error('BrowseFAILED')
+        console.error(err)
+      })
+  }
+
+  importConfig (e) {
+    if (this.state.newImport.path === '') {
+      alert(i18n.__('No file has been selected.'))
+      return false
+    }
+
+    try {
+      const externalConfig = JSON.parse(fs.readFileSync(this.state.newImport.path, 'utf-8'))
+      const { config } = this.state
+      config.hotkey = externalConfig.hotkey
+      config.ui = externalConfig.ui
+      config.editor = externalConfig.editor
+      config.preview = externalConfig.preview
+      config.blog = externalConfig.blog
+
+      this.setState({
+        config
+      })
+      this.handleSaveImport(e)
+    } catch (err) {
+      alert(i18n.__('File is not valid JSON.'))
+      return false
+    }
+  }
+
+  exportConfig (e) {
+    if (this.state.newExport.path === '') {
+      alert(i18n.__('Please choose a directory for export file.'))
+      return false
+    }
+    fs.writeFile(
+      `${this.state.newExport.path}/boostnote.config`,
+      JSON.stringify(this.state.config), (err) => {
+        if (err) {
+          return console.log(err)
```

Comment:
How about adding some kind of way to notice the user when saving is successful or failed? Because otherwise a think called [`Gulf of evaluation`](https://en.wikipedia.org/wiki/Gulf_of_evaluation) will happens.